src/events/subscriptions.js

```diff
@@ -7,34 +7,31 @@ var initialState = { block: {} };
 var subscriptions = assign({}, initialState);
 
 module.exports.onLogAdded = function (log) {
-  if (subscriptions[log.address] && subscriptions[log.address][log.topics[0]]) {
-    subscriptions[log.address][log.topics[0]].callback(log);
+  if (subscriptions[log.address]) {
+    subscriptions[log.address].callback(log);
   }
 };
 
 module.exports.onLogRemoved = function (log) {
-  if (subscriptions[log.address] && subscriptions[log.address][log.topics[0]]) {
+  if (subscriptions[log.address]) {
     log.removed = true;
-    subscriptions[log.address][log.topics[0]].callback(log);
+    subscriptions[log.address].callback(log);
   }
 };
 
 module.exports.getSubscriptions = function () {
   return subscriptions;
 };
 
-module.exports.addSubscription = function (contractAddress, eventSignature, token, callback) {
-  if (!subscriptions[contractAddress]) subscriptions[contractAddress] = {};
-  subscriptions[contractAddress][eventSignature] = { token: token, callback: callback };
+module.exports.addSubscription = function (contractAddress, token, callback) {
+  subscriptions[contractAddress] = { token: token, callback: callback };
 };
 
 module.exports.removeSubscription = function (token) {
   subscriptions = Object.keys(subscriptions).reduce(function (p, contractAddress) {
-    Object.keys(subscriptions[contractAddress]).forEach(function (eventSignature) {
-      if (subscriptions[contractAddress][eventSignature].token !== token) {
-        p[contractAddress][eventSignature] = subscriptions[contractAddress][eventSignature];
-      }
-    });
+    if (subscriptions[contractAddress].token !== token) {
+      p[contractAddress] = subscriptions[contractAddress];
```

Comment:
Unrelated to this PR, `p` is a _terrible_ variable name.

Reply:
Well, when we go to 1, we don't need it anymore!

Reply:
Ah, yes I have a bad habit of using `p` (...short for `previous`) for reduce's accumulator parameter... please feel free to rename these to something more descriptive when you encounter them :grimacing: